pkg/ebpf/dns_parser.go

```diff
@@ -0,0 +1,109 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"bytes"
+
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+const maxIPBufferSize = 200
+
+type dnsParser struct {
+	decoder *gopacket.DecodingLayerParser
+	layers  []gopacket.LayerType
+	payload *layers.DNS
+
+	// Cached translation object to reduce allocations
+	cached *translation
+}
+
+func newDNSParser() *dnsParser {
+	payload := &layers.DNS{}
+
+	stack := []gopacket.DecodingLayer{
+		&layers.Ethernet{},
+		&layers.IPv4{},
+		&layers.IPv6{},
+		&layers.UDP{},
+		payload,
+	}
+
+	return &dnsParser{
+		decoder: gopacket.NewDecodingLayerParser(layers.LayerTypeEthernet, stack...),
+		payload: payload,
+		cached:  new(translation),
+	}
+}
+
+func (p *dnsParser) Parse(data []byte) *translation {
+	err := p.decoder.DecodeLayers(data, &p.layers)
```

Comment:
Is `p.layers` reused between parse requests?  If so, does it need to be reset in some way ?

Reply:
`gopacket` does that. Their API is very confusing :/

Reply:
Can we add a comment that the library controls that? The more comments in this code the better - but can come in a later PR